### PR TITLE
#160617642 tagging articles

### PR DESCRIPTION
**Pull request by zaabu**
#### What does this PR do?
- This PR implements tagging functionality. This enables authors to create tags for the articles that they create.
- It also enables users to categorize articles by a certain tag.

#### Description of Task to be completed?

Tagging refers to the metadata that is assigned to a piece of content by the content creator and the readers/users of the content. Tagging allows for organization and or categorization of content.

#### How should this be manually tested?

- Enter a comma-separated string of words when creating an article and the words will be stored as tags
- Visit an article and you should be able to view tags for that article.
- Click the tags at the end of the article and you should be redirected to a page that shows articles under that tag.

#### Creating tags
screen shot 2018-11-21 at 18 12 24

#### Viewing tags on an article

screen shot 2018-11-21 at 18 12 45

#### Viewing articles under a given tag

screen shot 2018-11-21 at 18 12 55

### CODE REVIEW

File: src/actions/actionTypes.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ export const PASSWORD_RESET = "PASSWORD_RESET";
 export const NEW_PASSWORD = "NEW_PASSWORD";
 export const USER_NOT_FOUND = "USER_NOT_FOUND";
 export const REGISTER_USER = "REGISTER_USER";
+export const VIEW_ARTICLES_UNDER_TAG = "VIEW_ARTICLES_UNDER_TAG";
```
Comment (LehruAngela) [inline src/actions/actionTypes.js]: Declare your action in the ACTION_TYPE dict

File: src/views/articles/viewArticle.js [hunk 2]
```diff
@@ -54,7 +55,19 @@ const ViewOneArticle = ({ article, emailShare, handleEmail }) => {
...
+                <ul className="tags" >
+                  {!article.tags ? "" : article.tags.map(tag => (
+                    
+                    <Link to={`/view-articles-under-tag/${tag}`} key={tag}>
+                      <li className="d-inline tag-name"
+                        value={tag}>
+                        {tag}
+                      </li>
+                    </Link>
+
```
Comment (LehruAngela) [inline src/views/articles/viewArticle.js:77]: Remove empty lines of code, ensure you do that in all your files

File: src/assets/articleAssets/demo2.scss [hunk 3]
```diff
@@ -499,3 +499,42 @@ a:hover {
 }  
   
   
+
```
Comment (LehruAngela) [inline src/assets/articleAssets/demo2.scss]: Separate the tag styles in a different file, and rename `demo2.scss` to a more descriptive file name

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
Add a tag title or something along those lines on the `view all articles for a tag page` so as to identify which tag under which the articles are being rendered. @zaabu, Also write tests

File: src/assets/articleAssets/demo2.scss (hunk 3 above)
Comment (zaabu) [reply src/assets/articleAssets/demo2.scss]: I have resolved this @LehruAngela

File: src/views/articles/viewArticle.js (hunk 2 above)
Comment (zaabu) [reply src/views/articles/viewArticle.js:77]: I have resolved this @LehruAngela

File: src/actions/actionTypes.js (hunk 1 above)
Comment (zaabu) [reply src/actions/actionTypes.js]: I have resolved this @LehruAngela  by declaring VIEW_ARTICLES_UNDER_TAG under ACTION_TYPE

File: src/assets/articleAssets/tags.scss [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  @media screen and (min-width:772px) and (max-width: 923px)  {
+    .tag-title-container {
+      margin-left:80px;
+    }
+  }
+
+
+
+
+
```
Comment (LehruAngela) [inline src/assets/articleAssets/tags.scss]: Remove empty lines of code, and leave one empty line at end of file, ensure to do this in all files

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
Rebase with develop @zaabu

File: src/assets/articleAssets/tags.scss (hunk 4 above)
Comment (zaabu) [reply src/assets/articleAssets/tags.scss]: I have resolved this @LehruAngela

**Pull request merged**